src/libs/antares/study/parameters.cpp

```diff
@@ -205,6 +207,18 @@ void Parameters::resetAdqPatchParameters()
     adqPatch.localMatching.setToZeroOutsideOutsideLinks = true;
 }
 
+void Parameters::resetPlayedYears(uint nbOfYears)
+{
+    // Set the number of years
+    nbYears = std::min(nbOfYears, maximumMCYears);
+
+    // Reset the filter
+    delete[] yearsFilter;
+    yearsFilter = new bool[nbYears];
+    for (uint i = 0; i != nbYears; ++i)
+        yearsFilter[i] = true;
+}
+
```

Comment:
How about creating a dedicated class here ? With a dedicated API, so that we could add unit tests

```cpp
class Playlist {
public:
  void initialize(uint nbYears);
  bool setYearPlayed(uint y, bool played);
  bool isYearPlayed(uint y) const;
private:
  std::vector<bool> mYears;
};